Nano node software does not build when NANO_TEST is not set

### Summary

Nano node software does not build when NANO_TEST is not set.
You may need to build from scratch to reproduce the problem.

I built with these options to reproduce it:
```
cmake -G "Unix Makefiles" -DNANO_GUI=ON -DNANO_TEST=0 -DCMAKE_BUILD_TYPE=Debug ../nano-node
```

There is tail of the build log showing the problem.
```
[ 96%] Building CXX object nano/node/CMakeFiles/node.dir/wallet.cpp.o
[ 96%] Building CXX object nano/node/CMakeFiles/node.dir/websocket.cpp.o
[ 96%] Building CXX object nano/node/CMakeFiles/node.dir/websocketconfig.cpp.o
[ 96%] Building CXX object nano/node/CMakeFiles/node.dir/write_database_queue.cpp.o
[ 96%] Linking CXX static library libnode.a
make[3]: Leaving directory '/fast/buildbug/build'
[ 96%] Built target node
make[3]: Entering directory '/fast/buildbug/build'
make[3]: Entering directory '/fast/buildbug/build'
Scanning dependencies of target qt
Scanning dependencies of target nano_node
make[3]: Leaving directory '/fast/buildbug/build'
make[3]: Entering directory '/fast/buildbug/build'
[ 96%] Building CXX object nano/nano_node/CMakeFiles/nano_node.dir/daemon.cpp.o
[ 97%] Building CXX object nano/nano_node/CMakeFiles/nano_node.dir/entry.cpp.o
make[3]: Leaving directory '/fast/buildbug/build'
make[3]: Entering directory '/fast/buildbug/build'
[ 97%] Building CXX object CMakeFiles/qt.dir/nano/qt/qt.cpp.o
[ 97%] Linking CXX executable ../../nano_node
[ 98%] Linking CXX static library libqt.a
make[3]: Leaving directory '/fast/buildbug/build'
[ 98%] Built target qt
make[3]: Entering directory '/fast/buildbug/build'
make[3]: Entering directory '/fast/buildbug/build'
[ 98%] Generating qrc_resources.cpp
Scanning dependencies of target qt_system
Scanning dependencies of target nano_wallet
make[3]: Leaving directory '/fast/buildbug/build'
make[3]: Entering directory '/fast/buildbug/build'
[ 98%] Building CXX object CMakeFiles/qt_system.dir/nano/qt_system/entry.cpp.o
make[3]: Leaving directory '/fast/buildbug/build'
make[3]: Entering directory '/fast/buildbug/build'
[ 98%] Building CXX object CMakeFiles/nano_wallet.dir/nano/nano_wallet/plat/default/icon.cpp.o
[100%] Building CXX object CMakeFiles/nano_wallet.dir/qrc_resources.cpp.o
[100%] Building CXX object CMakeFiles/nano_wallet.dir/nano/nano_wallet/entry.cpp.o
[100%] Linking CXX executable qt_system
make[3]: Leaving directory '/fast/buildbug/build'
[100%] Built target nano_node
/usr/bin/ld: cannot find -ltest_common
collect2: error: ld returned 1 exit status
make[3]: *** [CMakeFiles/qt_system.dir/build.make:113: qt_system] Error 1
make[3]: Leaving directory '/fast/buildbug/build'
make[2]: *** [CMakeFiles/Makefile2:607: CMakeFiles/qt_system.dir/all] Error 2
make[2]: *** Waiting for unfinished jobs....
[100%] Linking CXX executable nano_wallet
/usr/bin/ld: cannot find -ltest_common
collect2: error: ld returned 1 exit status
make[3]: *** [CMakeFiles/nano_wallet.dir/build.make:151: nano_wallet] Error 1
make[3]: Leaving directory '/fast/buildbug/build'
make[2]: *** [CMakeFiles/Makefile2:524: CMakeFiles/nano_wallet.dir/all] Error 2
make[2]: Leaving directory '/fast/buildbug/build'
make[1]: *** [Makefile:152: all] Error 2
make[1]: Leaving directory '/fast/buildbug/build'
make: *** [Makefile:57: build] Error 2
```

### Node version

e3563a95db40e499cc4381b91eef78b15f89113b

### Build details

manually built

### OS and version

ubuntu 20.04

### Steps to reproduce the behavior

Build with options:
```
cmake -G "Unix Makefiles" -DNANO_GUI=ON -DNANO_TEST=0 -DCMAKE_BUILD_TYPE=Debug ../nano-node
```

### Expected behavior

To build

### Actual behavior

Does not build

### Possible solution

_No response_

### Supporting files

_No response_

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -639,7 +639,6 @@ if(NANO_GUI OR RAIBLOCKS_GUI)
     node
     secure
     nano_lib
-    test_common
     libminiupnpc-static
     Qt5::Gui
     Qt5::Widgets)

